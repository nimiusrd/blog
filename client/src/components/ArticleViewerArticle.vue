<template>
    <div class="article">
        <div class="author">
            author: {{ author }}
        </div>
        <div class="posted-at">
            posted at {{ postedAt }}
        </div>
        <div class="updated-at">
            updated at {{ updatedAt }}
        </div>
        <div class="title">
            {{ title }}
        </div>
        <div class="content ql-editor" v-html="content" />
    </div>
</template>

<script lang="ts">
import Vue from 'vue'

const ArticleViewerArticle = Vue.extend({
    created() {
        this.getArticle()
    },
    data() {
        return {
            author: '',
            content: '',
            postedAt: '',
            title: '',
            updatedAt: ''
        }
    },
    methods: {
        getArticle() {
            const vm = this
            const id = this.$route.params.id

            fetch(
                `/api/articles/${id}`,
                {
                    method: 'GET'
                }
            )
                .then(res => res.json())
                .then(json => {
                    vm.author = json.author
                    vm.content = json.content
                    vm.postedAt = new Date(json['postedAt']).toDateString()
                    vm.updatedAt = new Date(json['updatedAt']).toDateString()
                    vm.title = json.title
                })
                .catch(reason => {
                    console.log(reason)
                })
        }
    }
})

export default ArticleViewerArticle
</script>

<style scoped>
@import '~quill/dist/quill.core.css';

.article {
    display: grid;
    grid-template:
        'author posted-at updated-at'
        'title title title'
        'content content content';
}
.author {
    grid-area: author;
    padding: 8px;
}
.posted-at {
    grid-area: posted-at;
    padding: 8px;
}
.updated-at {
    grid-area: updated-at;
    padding: 8px;
}
.title {
    font-size: 48px;
    font-weight: 600;
    grid-area: title;
    padding: 24px;
}
.content {
    grid-area: content;
}
</style>
