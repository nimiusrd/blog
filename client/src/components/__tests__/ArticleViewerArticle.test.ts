import ArticleViewerArticle from '@/components/ArticleViewerArticle.vue'
import {shallowMount} from '@vue/test-utils'

test('snapshot', () => {
    const wrapper = shallowMount(ArticleViewerArticle, {
        methods: {
            getArticle() {
                this.$data.author = 'author'
                this.$data.content = 'content'
                this.$data.postedAt = '2019-09-28T06:34:56+00:00'
                this.$data.updatedAt = '2019-09-28T06:34:56+00:00'
                this.$data.title = 'title'
            }
        }
    })

    expect(wrapper.html()).toMatchSnapshot()
})
