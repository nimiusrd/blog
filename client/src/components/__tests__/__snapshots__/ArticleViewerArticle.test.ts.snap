// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
"<div class=\\"article\\"><div class=\\"author\\">
        author: author
    </div> <div class=\\"posted-at\\">
        posted at Sat Sep 28 2019
    </div> <div class=\\"updated-at\\">
        updated at Sat Sep 28 2019
    </div> <div class=\\"title\\">
        title
    </div> <div class=\\"content ql-editor\\">content</div></div>"
`;
